### PR TITLE
#1033 Make it easier to configure frame options header

### PR DESCRIPTION
**Pull request by adaur**

### CODE REVIEW

File: header.php [hunk 1]
```diff
@@ -19,8 +19,9 @@
 // Send the Content-type header in case the web server is setup to send something else
 header('Content-type: text/html; charset=utf-8');
 
-// Prevent site from being embedded in a frame
-header('X-Frame-Options: deny');
+// Prevent site from being embedded in a frame, if that's not what we want
+if (!defined('FORUM_EMBEDDED_IFRAME'))
+	header('X-Frame-Options: deny');
```
Comment (franzliedke) [inline header.php]: Sorry for criticizing all of these...

What I had in mind was something along these lines:
~~~php
$frame_options = defined('FORUM_FRAME_OPTIONS') ? FORUM_FRAME_OPTIONS : 'deny';
header('X-Frame-Options: '.$frame_options');
~~~

Thanks for taking care of all these! =)

**Comment by adaur** [comment]
What I have in mind is something like FORUM_BEHIND_PROXY. If FORUM_BEHIND_PROXY is set, then we know we have to deal with a reverse proxy and get_remote_address() will react differently.

Here, if FORUM_EMBEDDED_IFRAME is set, then we know the forum will be embedded via iframe and we have to disable the header.

We should update the doc to reflect this change, though.

**Comment by franzliedke** [comment]
But instead of disabling the header, we could also set it to `sameorigin`, which still gives us security against clickjacking attacks from other domains. Only when embedding in other domains, disabling it makes sense (although I think a whitelist of domains can be specified, too).

**Comment by franzliedke** [comment]
So, if you want to embed the forum in the same domain, all you'd have to do is put this line in your `config.php`:

~~~php
define('FORUM_FRAME_OPTIONS', 'sameorigin');
~~~

**Pull request closed without merging**